src/components/parcels/ParcelForm.svelte

```diff
@@ -0,0 +1,313 @@
+  export let mode: 'create' | 'edit' = 'create';
+  export let user: User | null;
+
+  let hasPermission: boolean = false;
+  let pageSubtitle: string = '';
+  let pageTitle: string = '';
+  let parcelModified: boolean = false;
+  let parcelChannelDictionaryId: number | null = initialParcelChannelDictionaryId;
+  let parcelCommandDictionaryId: number | null = initialParcelCommandDictionaryId;
+  let parcelCreatedAt: string | null = initialParcelCreatedAt;
+  let parcelName: string = initialParcelName;
+  let parcelId: number | null = initialParcelId;
+  let parcelOwner: UserId = initialParcelOwner;
+  let parcelSequenceAdaptationId: number | null = initialSequenceAdaptationId;
+  let permissionError = 'You do not have permission to edit this parcel.';
+  let saveButtonClass: 'primary' | 'secondary' = 'primary';
+  let saveButtonText: string = '';
+  let saveButtonEnabled: boolean = false;
+  let savedParcelChannelDictionaryId: number | null = parcelChannelDictionaryId;
+  let savedParcelCommandDictionaryId: number | null = parcelCommandDictionaryId;
+  let savedParcelName: string = parcelName;
+  let savedParameterDictionaryIds: Record<number, boolean> = {};
+  let savedSequenceAdaptationId: number | null = parcelSequenceAdaptationId;
+  let savingParcel: boolean = false;
+  let selectedParmeterDictionaries: Record<number, boolean> = {};
+
+  $: selectedParmeterDictionaries = savedParameterDictionaryIds = $parcelToParameterDictionaries.reduce(
+    (prevBooleanMap: Record<number, boolean>, parcelToParameterDictionary: ParcelToParameterDictionary) => {
+      return {
+        ...prevBooleanMap,
+        [parcelToParameterDictionary.parameter_dictionary_id]: true,
+      };
+    },
+    {},
+  );
+
+  $: saveButtonClass = parcelModified && saveButtonEnabled ? 'primary' : 'secondary';
+  $: saveButtonEnabled = parcelCommandDictionaryId !== null && parcelName !== '';
+  $: parcelModified =
+    parcelChannelDictionaryId !== savedParcelChannelDictionaryId ||
+    parcelCommandDictionaryId !== savedParcelCommandDictionaryId ||
+    parcelName !== savedParcelName ||
+    parcelSequenceAdaptationId !== savedSequenceAdaptationId ||
+    didParameterDictionariesChange(selectedParmeterDictionaries);
+
+  $: {
+    hasPermission =
+      mode === 'edit'
+        ? featurePermissions.parcels.canUpdate(user, { owner: parcelOwner })
+        : featurePermissions.parcels.canCreate(user);
+    permissionError = `You do not have permission to ${mode === 'edit' ? 'edit this' : 'create a'} parcel.`;
+    pageTitle = mode === 'edit' ? 'Parcel' : 'New Parcel';
+    pageSubtitle = mode === 'edit' ? savedParcelName : '';
+    saveButtonText = mode === 'edit' && !parcelModified ? 'Saved' : 'Save';
+  }
+
+  /**
+   * selectedParameterDictionaries keeps track of false values for the table while the saved list doesn't so
+   * we need to do a custom comparison.
+   */
+  function didParameterDictionariesChange(parameterDictionaryMap: Record<number, boolean>): boolean {
```

Comment:
Is this function needed? If the `selectedParmeterDictionaries` object changes, that means it was modified, right?

Reply:
`selectedParameterDictionaries` keeps false values set when you uncheck a dictionary and the saved list doesn't, so we can't do a direct comparison between them. An example would be if you loaded a page with 1 parameter dictionary selected and then selected a second one and then unchecked that second one.